hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -885,6 +888,29 @@ public int minPipelineLimit(List<DatanodeDetails> dnList) {
     return Collections.min(pipelineCountList);
   }
 
+  @Override
+  public Collection<DatanodeDetails> getPeerList(DatanodeDetails dn) {
+    HashSet<DatanodeDetails> dns = new HashSet<>();
+    Preconditions.checkNotNull(dn);
+    Set<PipelineID> pipelines =
+        nodeStateManager.getPipelineByDnID(dn.getUuid());
+    if (!pipelines.isEmpty()) {
+      pipelines.forEach(id -> {
+        try {
+          Pipeline pipeline =
+              scmContext.getScm().getPipelineManager().getPipeline(id);
```

Comment:
We can cache the `PipelineManager` instance outside of `forEach` instead of making `scmContext.getScm().getPipelineManager()` call inside `forEach` every time.